src/components/EnterBarcode/EnterBarcodePage.tsx

```diff
@@ -21,31 +21,35 @@ const EnterBarcodePage = ({
   const [barcode, setBarcode] = useState<string>('');
 
   const handleEnterBarcode = () => {
-    if (!loading && data) {
-      const selectedPatient = data.patients.filter(
-        (patient) => patient.barcodeValue === barcode
-      );
-      if (selectedPatient.length > 0) {
-        // Found patient
-        const {
-          collectionPointId: {
-            id: patientCcpId,
-            eventId: { id: patientEventId },
-          },
-          id,
-        } = selectedPatient[0];
-        // Redirect to patient profile
-        history.push(
-          `/events/${patientEventId}/ccps/${patientCcpId}/patients/${id}`
+    if (barcode !== '') {
```

Comment:
this check doesn't seem necessary - if we only call `handleEnterBarcode` when the form is submitted (i.e. when `barcode` is not an empty string), then this condition is always true as we cannot submit the form when `barcode` is an empty string

Reply:
Sounds good - in the new commit, I performed the check within `onSubmit` rather than inside of `handleEnterBarcode` since handleEnterBarcode should simply be routing the user rather than worrying about empty strings

Reply:
Also, I prevented the reloading with `e.preventDefault()` - hopefully that's the correct way to do so